### PR TITLE
Support filtering using startsWith

### PR DESCRIPTION
**Pull request by alexashley**

### CODE REVIEW

File: go/v1beta1/storage/filtering/filtering.go [hunk 1]
```diff
@@ -114,6 +121,17 @@ func parseExpression(expression *expr.Expr) (*Query, error) {
...
 			},
 		}, nil
+	case overloads.StartsWith:
+		leftTerm, rightTerm, err := getSimpleExpressionTerms(leftArg, rightArg)
+		if err != nil {
+			return nil, err
+		}
+
+		return &Query{
+			Prefix: &Term{
```
Comment (alexashley) [inline go/v1beta1/storage/filtering/filtering.go:131]: This is the shortened version of Elasticsearch's [prefix query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-prefix-query.html#prefix-query-short-ex)

File: go/v1beta1/storage/filtering/filtering.go [hunk 2]
```diff
@@ -46,8 +47,14 @@ func parseExpression(expression *expr.Expr) (*Query, error) {
 	function := expression.GetCallExpr().GetFunction()
 
 	// Determine if left and right side are final and if so formulate query
-	leftArg := expression.GetCallExpr().Args[0]
-	rightArg := expression.GetCallExpr().Args[1]
+	var leftArg, rightArg *expr.Expr
+	if len(expression.GetCallExpr().Args) == 2 {
```
Comment (alexashley) [inline go/v1beta1/storage/filtering/filtering.go:51]: For an expression `a == b`, `a` and `b` are treated as arguments to the `_==_` operator; but in an expression `a.startsWith(b)`, `a` is the target/receiver and `b` is the argument.

Comment (mrparkers) [reply go/v1beta1/storage/filtering/filtering.go:51]: can you add this explanation as a comment here?  we can remove it later once we figure out a better way for this

File: test/v1beta1/occurrence_test.go [hunk 3]
```diff
@@ -201,6 +214,36 @@ func TestOccurrence(t *testing.T) {
...
+				{
+					name:   "match resource.uri startsWith partially",
+					filter: `"resource.uri".startsWith("https://docker.io/library/alpine")`,
+					expected: []*grafeas_go_proto.Occurrence{
+						alpineVulnerabilityOccurrence,
+						secondAlpineVulnerabilityOccurrence,
+					},
+				},
+				{
+					name:     "match kind startsWith no match",
```
Comment (mrparkers) [inline test/v1beta1/occurrence_test.go:243]: this reads a little funky, can this wording be cleaned up?  maybe `match nothing via startsWith` or something like that

**Review by mrparkers — APPROVED** [review]

**Pull request merged**